nlp_apps.ipynb

```diff
@@ -479,7 +479,27 @@
     "\n",
     "As with any Natural Language Processing problem, it is prudent to do some text pre-processing and clean our data before we start building our model. Remember that all the papers are signed as 'Publius', so we can safely remove that word, since it doesn't give us any information as to the real author.\n",
     "\n",
-    "NOTE: Since we are only removing a single word from each paper, this step can be skipped. We add it here to show that processing the data in our hands is something we should always be considering. Oftentimes pre-processing the data in just the right way is the difference between a robust model and a flimsy one."
+    "NOTE: Since we are only removing a single word from each paper, this step can be skipped. We add it here to show that processing the data in our hands is something we should always be considering. Oftentimes pre-processing the data in just the right way is the difference between a robust model and a flimsy one. \n",
+    "\n",
+    "For example,after getting the words, we need the root form rather than the original word. For eg. the words \"support\", \"supportive\", \"supporting\" have the root word \"support\". It is better to have a single root word for multiple different words with similar meaning. It reduces the dimension of data. There are many stemming algorithms in NLP, popular ones being the porter stemmer and snowball stemmer.\n",
```

Comment:
This paragraph needs a bit of work:

a) Add a space after the first comma. Remember that after punctuation you need to have a space (at least, most of the time).

b) "For eg." - Write the word 'example' out instead of 'eg.'

c) "It reduces the dimension of data" - The word 'dimension' needs to be plural, I believe.

d) "porter stemmer and snowball stemmer" - Algorithm names need to be capitalized.